### ISSUE TITLE
azurerm_network_interface doc page says ip_configuration is Optional

### ISSUE THREAD

**Issue by ggriffiths** [hashicorp/terraform#12184]
### Terraform Version
Terraform v0.8.7

### Affected Resource(s)
azurerm_network_interface

### Terraform Configuration Files
```hcl
resource "azurerm_resource_group" "test" {
  name     = "acceptanceTestResourceGroup1"
  location = "West US"
}

resource "azurerm_virtual_network" "test" {
  name                = "acceptanceTestVirtualNetwork1"
  address_space       = ["10.0.0.0/16"]
  location            = "West US"
  resource_group_name = "${azurerm_resource_group.test.name}"
}

resource "azurerm_subnet" "test" {
  name                 = "testsubnet"
  resource_group_name  = "${azurerm_resource_group.test.name}"
  virtual_network_name = "${azurerm_virtual_network.test.name}"
  address_prefix       = "10.0.2.0/24"
}

resource "azurerm_network_interface" "test" {
  name                = "acceptanceTestNetworkInterface1"
  location            = "West US"
  resource_group_name = "${azurerm_resource_group.test.name}"

  #no ip_configuration block

  tags {
    environment = "staging"
  }
}
```

### Debug Output
https://gist.github.com/ggriffiths/5754779a8c6e729af8447383a7e58f82

### Panic Output
n/a

### Expected Behavior
Documentation should say "Required" for this block:
https://www.terraform.io/docs/providers/azurerm/r/network_interface.html

### Actual Behavior
Documentation says "Optional" for this block.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform plan`

### Important Factoids
None, this is with a standard Azure trial account 

### References
None

**Comment by ggriffiths** [comment]
New to the project, but found what I think is a minor issue in the docs. I might be wrong, feel free to close if so.

**Comment by stack72** [comment]
Closed via #12185
